packages/core/src/config/parse.ts

```diff
@@ -262,11 +277,47 @@ export const parseArrayElements = (
  * @param props standard VariableHandler props. See ./iterator.ts for more information.
  * @returns
  */
-export const parseInplaceArray: VariableHandler<ParsedConfigVariable> = (
-  props: VariableHandlerProps<ParsedConfigVariable>
-) => {
+export const parseInplaceArray: VariableHandler<
+  UserConfigVariable,
+  Array<ParsedConfigVariable>
+> = (
+  props: VariableHandlerProps<UserConfigVariable, Array<ParsedConfigVariable>>
+): Array<ParsedConfigVariable> => {
   const { storageObj, variable, storageTypes, nestedSlotOffset, dereferencer } =
     props
+
+  if (!Array.isArray(variable)) {
+    throw new InputError(
+      `Expected array for ${storageObj.label} but got ${typeof variable}`
+    )
+  }
+
+  // array object types come in the format: t_array(t_<type>)<size>_storage)
+  // when nested, the format is repeated: t_array(t_array(t_<type>)<size>_storage)<size>_storage)
+  // So to get the size of the array, we split on the ')' character, remove the first element (which is the type),
+  // remove the _storage suffix, and parse the remaining element as an integer.
+  let stringSizes = storageObj.type.split(')')
+  stringSizes = stringSizes.map((el) => el.replace('_storage', ''))
+  stringSizes.shift()
+  const sizes = stringSizes.map((el) => parseInt(el, 10))
```

Comment:
Probably cleaner to use a regex here. [This is how OZ does a similar thing](https://github.com/OpenZeppelin/openzeppelin-upgrades/blob/master/packages/core/src/scripts/migrate-oz-cli-project.ts#L330). Not necessary to do now